oci/private/image_index.sh.tpl

```diff
@@ -15,17 +15,18 @@ function add_image() {
     local image_path="$1"
     local output_path="$2"
 
-    local manifests=$("${YQ}" eval '.manifests[]' "${image_path}/index.json")
+    local manifests=$("${JQ}" -c '.manifests[]' "${image_path}/index.json")
 
     for manifest in "${manifests}"; do
-        local manifest_blob_path=$("${YQ}" '.digest | sub(":"; "/")' <<< ${manifest})
-        local config_blob_path=$("${YQ}" '.config.digest | sub(":"; "/")' "${image_path}/blobs/${manifest_blob_path}")
-
-        local platform=$("${YQ}" --output-format=json '{"os": .os, "architecture": .architecture, "variant": .variant, "os.version": .["os.version"], "os.features": .["os.features"]} | with_entries(select( .value != null ))' "${image_path}/blobs/${config_blob_path}")
-
-        platform="${platform}" \
-        manifest="${manifest}" \
-        "${YQ}" --inplace --output-format=json '.manifests += [env(manifest) + {"platform": env(platform)}]' "${output_path}/manifest_list.json"
+        local manifest_blob_path=$("${JQ}" -r '.digest | sub(":"; "/")' <<< ${manifest})
+        local config_blob_path=$("${JQ}" -r '.config.digest | sub(":"; "/")' "${image_path}/blobs/${manifest_blob_path}")
+
+        local platform=$("${JQ}" -c '{"os": .os, "architecture": .architecture, "variant": .variant, "os.version": .["os.version"], "os.features": .["os.features"]} | with_entries(select( .value != null ))' "${image_path}/blobs/${config_blob_path}")
+        "${JQ}" --argjson platform "${platform}" \
+                --argjson manifest "${manifest}" \
+                '.manifests |= [$manifest + {"platform": $platform}]'\
```

Comment:
Per [discussion in the "docker" channel](https://bazelbuild.slack.com/archives/CA3NW13MH/p1718836203461049) of the "Bazel" Slack workspace, this attempt to append to the "manifests" field doesn't work as intended.

Instead, I think we need the following, as suggested in [_jq_'s documentation on _Complex assignments_](https://jqlang.github.io/jq/manual/#complex-assignments):
```shell
'.manifests |= . + [$manifest + {"platform": $platform}]'\
```

Reply:
Please see #633 for a proposed correction.